app/controllers/email/VerifyPasscodeController.scala

```diff
@@ -21,46 +21,68 @@ import config.{AppConfig, ErrorHandler}
-import services.{EmailVerificationService}
+import services.EmailVerificationService
 import uk.gov.hmrc.http.HeaderCarrier
 import uk.gov.hmrc.play.HeaderCarrierConverter
+import views.html.email.PasscodeView
 
 import scala.concurrent.{ExecutionContext, Future}
 
 @Singleton
-class VerifyPasscodeController @Inject()(val emailVerificationService: EmailVerificationService,
-                                         val errorHandler: ErrorHandler)
+class VerifyPasscodeController @Inject()(emailVerificationService: EmailVerificationService,
+                                         errorHandler: ErrorHandler,
+                                         passcodeView: PasscodeView)
                                         (implicit val appConfig: AppConfig,
-                                      mcc: MessagesControllerComponents,
-                                      authComps: AuthPredicateComponents,
-                                      inFlightComps: InFlightPredicateComponents) extends BaseController {
+                                         mcc: MessagesControllerComponents,
+                                         authComps: AuthPredicateComponents,
+                                         inFlightComps: InFlightPredicateComponents) extends BaseController {
 
   implicit val ec: ExecutionContext = mcc.executionContext
 
   def emailShow: Action[AnyContent] = (blockAgentPredicate andThen inFlightEmailPredicate) { implicit user =>
     if (appConfig.features.emailPinVerificationEnabled()) {
       extractSessionEmail match {
-        case Some(email) => Ok("") //TODO
+        case Some(email) => Ok(passcodeView(email, PasscodeForm.form, contactPrefJourney = false))
         case _ => Redirect(routes.CaptureEmailController.show())
       }
     } else {
       NotFound(errorHandler.notFoundTemplate(user))
     }
   }
 
-  def emailSendVerification: Action[AnyContent] = blockAgentPredicate { implicit user =>
-
+  def emailSubmit: Action[AnyContent] = (blockAgentPredicate andThen inFlightEmailPredicate) { implicit user =>
     if (appConfig.features.emailPinVerificationEnabled()) {
       extractSessionEmail match {
-        case Some(email) => Ok("") //TODO
-
+        case Some(email) => PasscodeForm.form.bindFromRequest().fold(
+          error => {
+            BadRequest(passcodeView(email, error, contactPrefJourney = false))
+          },
+          passcode => {
+            Ok("Success") // TODO
+          }
+        )
         case _ => Redirect(routes.CaptureEmailController.show())
       }
     } else {
-        NotFound(errorHandler.notFoundTemplate(user))
+      NotFound(errorHandler.notFoundTemplate(user))
+    }
+  }
+
+  def emailSendVerification: Action[AnyContent] = blockAgentPredicate.async { implicit user =>
+    if (appConfig.features.emailPinVerificationEnabled()) {
+      extractSessionEmail match {
+        case Some(email) => emailVerificationService.isEmailVerified(email).map {
```

Comment:
Why are we calling isEmailVerified here? The "request-passcode" endpoint would return a 409 if the email is already verified so we could deal with it that way